Binary files linux-2.4.27.orig/net/bridge/.br_if.c.swp and linux-2.4.27/net/bridge/.br_if.c.swp differ
diff -urN linux-2.4.27.orig/net/bridge/br_if.c linux-2.4.27/net/bridge/br_if.c
--- linux-2.4.27.orig/net/bridge/br_if.c	2008-06-25 23:11:06.000000000 +0300
+++ linux-2.4.27/net/bridge/br_if.c	2008-06-25 23:14:38.000000000 +0300
@@ -265,15 +265,8 @@
 	int i = 0;
 
 	for (dev = dev_base; i < num && dev != NULL; dev = dev->next) {
-		if (dev->hard_start_xmit == br_dev_xmit) {
-		if (user_buf) {
-			if (copy_to_user(indices+i, &br->dev.ifindex, sizeof (br->dev.ifindex)))
-				return -EFAULT;
-			i++;
- 		} else {
-			indices[i++] = br->dev.ifindex;
-		}
-		}
+		if (dev->hard_start_xmit == br_dev_xmit)
+			indices[i++] = dev->ifindex;
 	}
 
 	return i;
